boards: frdm_mcxc444: Add usb support

NeilChen93 opened the pull request:
Support usb for NXP frdm_mcxc444 board

danieldegrasse commented:
Please take a look at the CI error

danieldegrasse commented:
@NeilChen93 PR needs rebase to clear conflicts

mmahadevan108 commented:
@NeilChen93 , kindly fix merge conflicts. Also please add information on how this was tested.

mmahadevan108 commented:
@NeilChen93, please update the 4.1 release notes to indicate this support was added.

kartben commented:
@NeilChen93 ping?

NeilChen93 committed:
samples: usb: Exclude frdm_mcxc444 platform from dfu sample

frdm_mcxc444 does not support MCUboot. Exclude the platform from dfu
sample.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

danieldegrasse commented:
@NeilChen93 thanks for rebasing- there is unfortunately another merge conflict :/, it needs another rebase

henrikbrixandersen commented:
Please rebase on `main` and resolve any merge conflicts.

mmahadevan108 commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -101,6 +101,7 @@ Boards & SoC Support
 
   * All HWMv1 board name aliases which were added as deprecated in v3.7 are now removed
     (:github:`82247`).
+  * Enabled USB on NXP ``frdm_mcxc444``
```
Please remove this change as the whole release notes for 4.1 is under discussion about needs to be included.

mmahadevan108 replied:
A rebase is required on this PR. So might as well remove it.

NeilChen93 committed:
boards: frdm_mcxc444: Add usb support

Support usb for NXP frdm_mcxc444 board.
Test it using samples/subsys/usb/console.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
doc: release-notes-4.1: Update 4.1 note

Remove USB, RTC enable note for NXP frdm_mcxn236

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

mmahadevan108 commented:
@NeilChen93 , was this tested with the new USB Next subsystem?
